### ISSUE TITLE
connections/cursor.js test failure in v2.0.x

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#3999]
While running `test/rql_test/test-runner` on 07fe398b95077f12f7c198c38662b99485b48028:
```
== Starting: connections/cursor.js (T+ 33.0 sec)
>>> Failed connections/cursor.js with exit code 1 after 14.1 sec (T+ 47.2 sec)

Inserting 1430 rows of data: . . . . . . . . . . . . . . . Using RethinkDB client from: /home/ssd3/daniel/rethinkdb/build/packages/js
Testing for 1430
First cursor was closed
[TypeError: Cannot read property 'length' of undefined]
TypeError: Cannot read property 'length' of undefinedTypeError: Cannot read property 'length' of undefined
    at _this._closeCb (/home/ssd3/daniel/rethinkdb/build/packages/js/cursor.js:220:33)
    at tryCatch1 (/home/ssd3/daniel/rethinkdb/build/packages/js/node_modules/bluebird/js/main/util.js:43:21)
    at Function.Promise$_successAdapter (/home/ssd3/daniel/rethinkdb/build/packages/js/node_modules/bluebird/js/main/nodeify.js:46:11)
    at tryCatch1 (/home/ssd3/daniel/rethinkdb/build/packages/js/node_modules/bluebird/js/main/util.js:43:21)
    at Promise$_callHandler [as _callHandler] (/home/ssd3/daniel/rethinkdb/build/packages/js/node_modules/bluebird/js/main/promise.js:649:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/ssd3/daniel/rethinkdb/build/packages/js/node_modules/bluebird/js/main/promise.js:663:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/home/ssd3/daniel/rethinkdb/build/packages/js/node_modules/bluebird/js/main/promise.js:826:14)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/home/ssd3/daniel/rethinkdb/build/packages/js/node_modules/bluebird/js/main/async.js:75:12)
    at Async$consumeFunctionBuffer (/home/ssd3/daniel/rethinkdb/build/packages/js/node_modules/bluebird/js/main/async.js:38:14)
    at process._tickCallback (node.js:415:13)
Done


<<< end Failed: connections/cursor.js
```

A couple of JS polyglot tests are also failing with the same error.

@deontologician any idea? I think this is a new problem.

**Comment by deontologician** [comment]
Probably a result of the recent cursor.close stuff. The test might need to be fixed, I'll check it out

**Comment by deontologician** [comment]
This should be fixed by #4002
